gateway/src/main/java/org/georchestra/gateway/security/ResolveHttpHeadersGeorchestraUserFilter.java

```diff
@@ -0,0 +1,146 @@
+        return ORDER;
+    }
+
+    public @Override Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
+        if (exchange.getRequest().getHeaders().containsKey("sec-mellon-name-id")) {
+            if (config.isCreateNonExistingUsersInLDAP()) {
+                String username = exchange.getRequest().getHeaders().get("sec-username").get(0);
+                Optional<GeorchestraUser> userOpt = map(username);
+
+                if (userOpt.isEmpty()) {
+                    try {
+                        String email = exchange.getRequest().getHeaders().get("sec-email").get(0);
+                        String longname = exchange.getRequest().getHeaders().get("sec-longname").get(0);
+                        String org = exchange.getRequest().getHeaders().get("sec-org").get(0);
+                        Account newAccount = AccountFactory.createBrief(username, null, longname.split(" ")[0],
+                                longname.split(" ")[1], email, "", "", "");
+                        newAccount.setPending(false);
+                        newAccount.setOrg(org);
+                        accountDao.insert(newAccount);
+                        roleDao.addUser(Role.USER, newAccount);
+                        userOpt = map(username);
+                    } catch (DuplicatedUidException e) {
+                        throw new IllegalStateException(e);
+                    } catch (DuplicatedEmailException e) {
+                        throw new IllegalStateException(e);
+                    } catch (DataServiceException e) {
+                        throw new IllegalStateException(e);
+                    }
+                }
+                GeorchestraUsers.store(exchange, userOpt.orElse(null));
+            } else {
+                GeorchestraUsers.store(exchange, map(exchange).orElse(null));
+            }
+            return chain.filter(exchange);
+        } else {
+            return chain.filter(exchange);
+        }
+    }
+
+    protected Optional<GeorchestraUser> map(String username) {
+        UserMapperImpl mapper = new UserMapperImpl();
+        mapper.setRoleDao(roleDao);
+        List<String> protectedUsers = Collections.emptyList();
+        UserRule rule = new UserRule();
+        rule.setListOfprotectedUsers(protectedUsers.toArray(String[]::new));
+        UsersApiImpl usersApi = new UsersApiImpl();
+        usersApi.setAccountsDao(accountDao);
+        usersApi.setMapper(mapper);
+        usersApi.setUserRule(rule);
+
+        Optional<GeorchestraUser> userOpt = usersApi.findByUsername(username);
+        if (userOpt.isPresent()) {
+            List<String> roles = userOpt.get().getRoles().stream().map(r -> r.contains("ROLE_") ? r : "ROLE_" + r)
+                    .collect(Collectors.toList());
+            if (roles.isEmpty()) {
+                roles.add("ROLE_USER");
+            }
+            userOpt.get().setRoles(roles);
+            userOpt.get().setOrganization("INRAE");
```

Comment:
this should not be hardcoded

Reply:
but then maybe we'd need to create an organization based on what Renater provides ? Will we have enough info for this ?

Reply:
Maybe we should put the users into a default organization to start with (relates to https://github.com/georchestra/georchestra/issues/4017)

Reply:
> Does Apache header contains organization ?

It could, I'd have liked to go further on this point and check the current geOrchestra service provider definition, unfortunately my renater account has been messed up somehow, and I am in the process of fixing it, which required to open a ticket at the Renater organization :(